src/Microsoft.AspNetCore.Authentication.AzureAd/AzureAdB2CInitializeOptions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Authentication.AzureAd
+{
+    internal class AzureAdB2CInitializeOptions : IInitializeOptions<AzureAdB2COptions>
+    {
+
+        public AzureAdB2CInitializeOptions() { }
+
+        public void Initialize(string name, AzureAdB2COptions options)
+        {
+            if (string.IsNullOrEmpty(options.Authority))
+            {
+                // Bind to only to any of the AzureAdB2C policy schemes
+                if (name == AzureDefaults.AzureAdB2CEditProfileAuthenticationScheme)
```

Comment:
Yeah Dan had to walk me through the scenario, basically there's some number of magic policies that B2C supports, there are 3 we currently support, and the idea is there's a single OIDC section in config we bind against, which is applied to each of the named schemes that map to the B2C policies (SignIn/ResetPassword/EditProfile)...  this is better than what the templates were doing before, but yeah its just a slightly cleaner hack :)